stores/listingCart.ts

```diff
@@ -0,0 +1,103 @@
+import { defineStore } from 'pinia'
+import { EntityWithId } from '~/components/rmrk/service/scheme'
+import { ComputedRef, set } from 'vue'
+import type { Prefix } from '@kodadot1/static'
+
+export type ListCartItem = {
+  id: string
+  name: string
+  urlPrefix: string
+  price: string
+  listPrice?: number
+  collection: EntityWithId
+}
+
+type ID = string
+
+interface State {
+  items: ListCartItem[]
+  owned: ListCartItem[]
+  chain: ComputedRef<Prefix>
+}
+
+const localStorage = useLocalStorage<ListCartItem[]>('listingCart', [])
+export const useListingCartStore = defineStore('listingCart', {
+  state: (): State => ({
+    items: localStorage.value,
+    owned: [],
+    chain: usePrefix().urlPrefix,
+  }),
+  getters: {
+    itemsInChain: (state): ListCartItem[] =>
+      state.items.filter((item) => item.urlPrefix === state.chain),
+    count() {
+      return this.itemsInChain.length
+    },
+    incompleteListPrices: (state) =>
+      state.items.filter((item) => !item.listPrice).length,
+  },
+  actions: {
+    getItem(id: ID) {
+      return this.items.find((item) => item.id === id)
+    },
+    isItemInCart(id: ID) {
+      return this.existInItemIndex(id) !== -1
+    },
+    existInItemIndex(id: string) {
+      return this.items.findIndex((item) => item.id === id)
+    },
+    setItem(payload: ListCartItem) {
+      const existInItemIndex = this.existInItemIndex(payload.id)
+      if (existInItemIndex === -1) {
+        this.items.push(payload)
+        localStorage.value = this.items
+      } else {
+        this.updateItem(payload)
+      }
+    },
+    addAllToCart() {
+      for (const item of this.owned) {
```

Comment:
please use ES6 method `forEach`

Reply:
> please use ES6 method `forEach`

is there a chance to have eslint for this one?

Reply:
<img width="1078" alt="image" src="https://github.com/kodadot/nft-gallery/assets/3810177/253aaf95-a93f-4f59-a0c8-0bd78e6ddbb6">
Should i also update the style_guide for this @yangwao

Reply:
> Should i also update the style_guide for this

make sense

Reply:
the last one for-of loop here 👀